src/rqt_plot/rosplot.py

```diff
@@ -94,25 +87,26 @@ class ROSData(object):
     Subscriber to ROS topic that buffers incoming data
     """
 
-    def __init__(self, topic, start_time):
+    def __init__(self, node, topic, start_time):
         self.name = topic
         self.start_time = start_time
         self.error = None
+        self.node = node
 
         self.lock = threading.Lock()
         self.buff_x = []
         self.buff_y = []
 
-        topic_type, real_topic, fields = get_topic_type(topic)
+        topic_type, real_topic, fields = get_topic_type(node, topic)
         if topic_type is not None:
             self.field_evals = generate_field_evals(fields)
-            data_class = roslib.message.get_message_class(topic_type)
-            self.sub = rospy.Subscriber(real_topic, data_class, self._ros_cb)
+            data_class = get_message_class(topic_type)
+            self.sub = node.create_subscription(data_class, real_topic, self._ros_cb)
```

Comment:
Might want to set the callback group to ReentrantCallbackGroup since the default is MutuallyExclusiveCallbackGroup and this is likely to be getting a lot of data.